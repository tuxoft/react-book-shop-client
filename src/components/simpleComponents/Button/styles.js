import styled from "styled-components";

export const Button = styled.button.attrs({ type: "button" })`
  padding: 0.5rem;
  border: 1px solid #f1f1f1;
  background-color: #fff;
  font-size: 1rem;
  border-radius: 5px;
  cursor: pointer;
  margin: auto 0;
`;
