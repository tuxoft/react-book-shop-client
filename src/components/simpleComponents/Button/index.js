import React from "react";
import PropTypes from "prop-types";
import * as styles from "./styles";

const Button = ({ children, ...restProps }) => (
  <styles.Button {...restProps}>{children}</styles.Button>
);

Button.propTypes = {
  onClick: PropTypes.func
};

export default Button;
