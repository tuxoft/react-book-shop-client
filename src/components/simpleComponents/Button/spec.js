import React from "react";
import Button from "./index";
import {ThemeProvider} from "styled-components";

import { shallow, mount } from "enzyme";

import shopTheme from '../../../constants/shopTheme';
it("renders text", () => {
  const button = mount(
      <ThemeProvider theme = {shopTheme}>
        <Button>Button</Button>
      </ThemeProvider>);
  expect(button.text()).toBe("Button");
});

it("renders text", () => {
  const onClickHandler = jest.fn();
  const button = mount(
      <ThemeProvider theme = {shopTheme}>
        <Button onClick={onClickHandler}>Button</Button>
      </ThemeProvider>);
  button.simulate("click");
  expect(onClickHandler).toHaveBeenCalled();
});

it("default button", () => {
  const button = shallow(
      <ThemeProvider theme = {shopTheme}>
        <Button>Button</Button>
      </ThemeProvider>);
  expect(button).toMatchSnapshot();
});

it("primary button", () => {
  const button = shallow(
   <ThemeProvider theme = {shopTheme}>
    <Button primary>Button</Button>
  </ThemeProvider>);
  expect(button).toMatchSnapshot();
});

it("small button", () => {
  const button = shallow(
      <ThemeProvider theme = {shopTheme}>
        <Button small>Button</Button>
      </ThemeProvider>);
  expect(button).toMatchSnapshot();
});
