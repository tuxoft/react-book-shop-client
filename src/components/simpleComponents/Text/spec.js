import React from "react";
import Button from "./index";
import { shallow, mount } from "enzyme";

it("renders text", () => {
  const button = mount(<Button>Button</Button>);
  expect(button.text()).toBe("Button");
});

it("renders text", () => {
  const onClickHandler = jest.fn();
  const button = mount(<Button onClick={onClickHandler}>Button</Button>);
  button.simulate("click");
  expect(onClickHandler).toHaveBeenCalled();
});

it("default button", () => {
  const button = shallow(<Button>Button</Button>);
  expect(button).toMatchSnapshot();
});

it("primary button", () => {
  const button = shallow(<Button primary>Button</Button>);
  expect(button).toMatchSnapshot();
});

it("small button", () => {
  const button = shallow(<Button small>Button</Button>);
  expect(button).toMatchSnapshot();
});
