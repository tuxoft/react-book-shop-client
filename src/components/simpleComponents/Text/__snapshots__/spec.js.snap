// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders bold 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "DARK_COLOR": "#2598c6",
        "GRAY_COLOR": "#A3AAB1",
        "THEME_COLOR": "#69c3e8",
      },
    }
  }
>
  <BoldText>
    <styled.div>
      <div
        className="sc-bwzfXH locxlO"
      >
        <styled.div>
          <div
            className="sc-bdVaJa eMFQSc"
          >
            Text
          </div>
        </styled.div>
      </div>
    </styled.div>
  </BoldText>
</ThemeProvider>
`;

exports[`renders colored 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "DARK_COLOR": "#2598c6",
        "GRAY_COLOR": "#A3AAB1",
        "THEME_COLOR": "#69c3e8",
      },
    }
  }
>
  <ColoredText>
    <styled.div>
      <div
        className="sc-htpNat ihLMWb"
      >
        <styled.div>
          <div
            className="sc-bdVaJa eMFQSc"
          >
            Text
          </div>
        </styled.div>
      </div>
    </styled.div>
  </ColoredText>
</ThemeProvider>
`;

exports[`renders coloredBold 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "DARK_COLOR": "#2598c6",
        "GRAY_COLOR": "#A3AAB1",
        "THEME_COLOR": "#69c3e8",
      },
    }
  }
>
  <ColoredBoldText>
    <styled.div>
      <div
        className="sc-bwzfXH locxlO"
      >
        <styled.div>
          <div
            className="sc-htpNat ihLMWb"
          >
            <styled.div>
              <div
                className="sc-bdVaJa eMFQSc"
              >
                Text
              </div>
            </styled.div>
          </div>
        </styled.div>
      </div>
    </styled.div>
  </ColoredBoldText>
</ThemeProvider>
`;

exports[`renders simple 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "DARK_COLOR": "#2598c6",
        "GRAY_COLOR": "#A3AAB1",
        "THEME_COLOR": "#69c3e8",
      },
    }
  }
>
  <SimpleText>
    <styled.div>
      <div
        className="sc-bdVaJa eMFQSc"
      >
        Text
      </div>
    </styled.div>
  </SimpleText>
</ThemeProvider>
`;

exports[`renders underlined 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "DARK_COLOR": "#2598c6",
        "GRAY_COLOR": "#A3AAB1",
        "THEME_COLOR": "#69c3e8",
      },
    }
  }
>
  <UnderlinedText>
    <styled.div>
      <div
        className="sc-bxivhb bFxTZR"
      >
        <styled.div>
          <div
            className="sc-bdVaJa eMFQSc"
          >
            Text
          </div>
        </styled.div>
      </div>
    </styled.div>
  </UnderlinedText>
</ThemeProvider>
`;
